Initial checkin of microprofile-config sample. Hammock updated to 2.1…

cchet opened the pull request:
This is a new sample for microprofile-config-api usage.
Please review and give me a feedback, if this sample is properly implemented and can be used as a sample for eclipse/microprofile-samples repository.

**How to build:** mvn clean install -Phammock
**How to run:**     java -DallowedUsers="het:mypass" -DlistAllConfigAllowed="true"  -jar .\config-1.0.0-SNAPSHOT-hammock.jar

**How to access:**
- http://localhost:8080/list/all
  Returns all config sources along with their set configurations, if **listAllConfigAllowed** configuration property is set to true
- http://localhost:8080/list/users
  Returns all users provided by **allowedUsers** configuration property
- http://localhost:8080/list/source/<name\>
  Returns all configurations of the given config source

This is the first draft of the microprofile-sample-config project.

**Changes:**
- Initial check in of new microprofile-sample-config
- Update hammock version to 2.1
- Had to add additional dependencies to new sample because ClassNotFoundExceptions occurred during startup
- Added capsule-maven-plugin (commented out) which could replace maven-shade-plugin 

Signed-off-by: Thomas Herzog <herzog.thomas81@gmail.com>

cchet committed:
Initial checkin of microprofile-config sample. Hammock updated to 2.1. Added dependencies because of ClassNotFoundExceptions. Added maen-capsule-plugin, but commented out, could replace maven-shade-plugin

Signed-off-by: Thomas Herzog <herzog.thomas81@gmail.com>

cchet commented:
I have recreated the pull request, because added a new README.adoc for the new sample and performed the changes suggested by Emily Jiang, which were

* change year in licenses
* remove beans.xml, because not used 
* signoff the commit

I have signed the commit and created and eclipse account, but still got the ip validation error. Sorry, but this is my first time contributing to eclipse, maybe I still did something wrong :)

cchet commented:
I have signed the ECA, but it still says IP-Valdiation failed.

Sorry for the inconvenience.

kenfinnigan commented:
Is herzog.thomas81@gmail.com the email associated with your ECA?

Also, it looks like CI failed because of a compile error

kenfinnigan commented:
Hah, looks like my accidental closing and re-open triggered the IP validation again

cchet commented:
Compiles on my local machine :)
Will take a look, when the build is done

cchet commented:
The issue with the build is the missing microprofile-config-api dependency for default profile.

But locally I get this error, when building the project with ``mvn install -DskipTests=true -B -V
``, which is the command used by Travis CI:

> [ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.12:check (default) on project config: Too many files with unapproved license: 5 See RAT report in: C:\Users\herzo\repositories\github\cchet\microprofile-samples\m
icroprofile-sample-config\target\rat-report.txt -> [Help 1]

How can I solve this ?

kenfinnigan commented:
Can you post what the content of that RAT report is?

cchet commented:
Here you go 

[rat-report.txt](https://github.com/eclipse/microprofile-samples/files/2316098/rat-report.txt)

kenfinnigan commented:
I think the files you've created need copyright/license headers similar to https://github.com/eclipse/microprofile-samples/blob/master/microprofile-sample-canonical/pom.xml

cchet committed:
- Updated to microprofile-config version, which is used by hammock
- Took 1.2 but hammock uses 1.2, therefore overwriting the former commit and using the proper microprofile-config version
- Added proper license header to source files

Signed-off-by: Thomas Herzog <herzog.thomas81@gmail.com>

cchet commented:
Should work now. 
Thanks for the help

cchet commented:
The tests fail for canonical, because of incompatibility of hammock provided weld-se-core and arquillian-arquillian-weld-embedded, whereby the test fail because of

`Caused by: java.lang.NoSuchMethodError: javax.enterprise.inject.spi.AfterBeanDiscovery.addBean()Ljavax/enterprise/inject/spi/configurator/BeanConfigurator;
	at org.jboss.weld.environment.se.WeldSEBeanRegistrant.registerWeldSEContexts(WeldSEBeanRegistrant.java:67)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:95)
	... 105 more`

It seems tho be weld 3.0.3.Final which uses CDI 2.0 API but not seems not to be supported by arquillian-weld-embedded.

I'm not sure how to solve this. Any ideas ?

cchet commented:
Ok, the problem is the javaee-api which does not contain cdi 2.0 spec.
Solved it by 

>     
            <dependency>
                <groupId>javax.enterprise</groupId>
                <artifactId>cdi-api</artifactId>
                <version>${version.cdi-api}</version>
                <scope>provided</scope>
            </dependency>

            <dependency>
                <groupId>javax</groupId>
                <artifactId>javaee-api</artifactId>
                <version>${version.javaee}</version>
                <scope>provided</scope>
            </dependency>

Whereby the cdi-api overlays the javax.enterprise classes of the javaee-api, because declared before the javaee-api.

Is there a better solution for solving this ?

cchet commented:
I think the first part should be to update the dependencies to a current version, because I ran into some issues. 
For instance, the wildfly-swarm tests are failing, because the fractions are not detected automatically anymore, which seems to be caused by the updated arquillian-universe dependency, which is used during compilation.

Therefore I close this PR and will provide the microprofile-config sample after updating the dependencies, as discussed in issue https://github.com/eclipse/microprofile-samples/issues/49.
